### PR TITLE
Fix comments in AggregateException.GetBaseException()

### PR DESCRIPTION
**Pull request by epsitec**
The public `<summary>` incorrectly stated that the method returned an `AggregateException`.
This causes confusion in the community.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/AggregateException.cs [hunk 1]
```diff
@@ -213,18 +213,19 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
         }
 
         /// <summary>
-        /// Returns the <see cref="AggregateException"/> that is the root cause of this exception.
+        /// Returns the <see cref="Exception"/> that is the root cause of this exception.
+        /// This will either be the root exception, or the first <see cref="AggregateException"/>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/AggregateException.cs:217]: What does "the root exception" mean here?

Comment (epsitec) [reply src/libraries/System.Private.CoreLib/src/System/AggregateException.cs:217]: The exception described as the _root cause_ in the initial comment. I've no idea how to phrase this in a clearer way.

**Comment by jkotas** [comment]
You may want to send the edit to https://github.com/dotnet/dotnet-api-docs/ repo instead or in addition to this PR. dotnet-api-docs repo has the actively maintained copy of the documentation for the most of the dotnet/runtime APIs, and it is the source of the content for https://learn.microsoft.com/. Updates in this repo won't show up at https://learn.microsoft.com/dotnet/api/system.aggregateexception.getbaseexception . We would like to fix this duplication, but we did not get to it yet.

**Comment by epsitec** [comment]
I also submitted another PR here https://github.com/dotnet/dotnet-api-docs/pull/10396

**Comment by skyoxZ** [comment]
Related: https://github.com/dotnet/runtime/issues/82853

It seems that this PR still not completely documents the unintuitive behavior.

**Comment by epsitec** [comment]
@skyoxZ I'd be glad if you could improve my message!

**Comment by skyoxZ** [comment]
I really don't want it to be documented. I prefer to keep it ambiguous if we can't fix the behavior.

**Comment by epsitec** [comment]
@jkotas can you approve the proposed edit?

File: src/libraries/System.Private.CoreLib/src/System/AggregateException.cs [hunk 2]
```diff
@@ -213,13 +213,14 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
...
+        /// that contains either multiple inner exceptions or no inner exceptions at all.
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/AggregateException.cs:218]: These lines should go into remarks rather than into summary.

Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/src/System/AggregateException.cs:218]: @epsitec do you want to make that fix so we can merge?

File: src/libraries/System.Private.CoreLib/src/System/AggregateException.cs [hunk 3]
```diff
@@ -213,13 +213,16 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
...
+        /// </remarks>
         public override Exception GetBaseException()
         {
-            // Returns the first inner AggregateException that contains more or less than one inner exception
-
-            // Recursively traverse the inner exceptions as long as the inner exception of type AggregateException and has only one inner exception
+            // Recursively traverse the inner exceptions as long as the inner exception of type
+            // AggregateException and has only one inner exception
```
Comment (danmoseley) [inline src/libraries/System.Private.CoreLib/src/System/AggregateException.cs:225]:
```suggestion
            // Recursively traverse the inner exceptions as long as the inner exception is of type
            // AggregateException and has exactly one inner exception
```

**Review by danmoseley — APPROVED** [review]

**Review by danmoseley — APPROVED** [review]
thanks

**Pull request merged**